### ISSUE TITLE
Log Completed Request: LogLevel off not work

### ISSUE THREAD

**Issue by rvojcik** [hashicorp/vault#24157]
**Describe the bug**
Default `log_requests_level` should be `off` by the documentation. By default Vault log all completed requests to stdout.
I tried to explicitly turn it off via config.json file `"log_requests_level": "off",`

My configuration 
```
    "log_level": "error",
    "log_requests_level": "off",

```

Vault still log every completed request even success ones as errors

```
Nov 16 10:06:55 vault3 vault[6098]: 2023-11-16T10:06:55.118+0100 [ERROR] core: completed_request: start_time="2023-11-16T10:06:55+01:00" duration=1ms client_id=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx client_address=xxxxxxxxxxxxxxxxxxxx status_code=200 request_path=/v1/secrets/data/xxxxxxxxxxxxxxxxxxxx request_method=GET
Nov 16 10:06:55 vault3 vault[6098]: 2023-11-16T10:06:55.122+0100 [ERROR] core: completed_request: start_time="2023-11-16T10:06:55+01:00" duration=1ms client_id=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx client_address=xxxxxxxxxxxxxxxxxx status_code=200 request_path=/v1/certs/data/xxxxxxxxxxxxxxxxxxxxxxx request_method=GET

```
**To Reproduce**
1. Run vault server with disabled log_requests_level and log_level error
2. Run read data from vault
3. See stdout/stderr logs from vault server

**Expected behavior**
See only error (log_level: error) and no completed_request is logged (log_requests_level: off)

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.15.2
* Vault CLI Version (retrieve with `vault version`): 1.15.2
* Server Operating System/Architecture: Debian 10 (Buster), AMD64, 

Vault server configuration file(s):
config.json
```json
{
     "api_addr": "https://xxxx:8200",
    "cluster_addr": "https://xxx:8201",
    "default_lease_ttl": "6h",
    "disable_mlock": true,
    "listener": {
        "tcp": {
            "address": "0.0.0.0:8200",
            "tls_cert_file": "xxxxx",
            "tls_key_file": "xxxxx"
        }
    },
    "log_level": "error",
    "log_requests_level": "off",
    "storage": {
        "raft": {
            "node_id": "vault3",
            "path": "/x/raft_storage"
        }
    },
    "telemetry": {
        "disable_hostname": true,
        "dogstatsd_addr": "localhost:8125",
        "prometheus_retention_time": "30s"
    },
    "ui": true

}
```

**Comment by delamart** [comment]
Can confirm I can't disable request logs which in Kubernetes shows every probe call

**Comment by adcsikos** [comment]
Same issue in my side as well! With vault verison:1.15.1-ubi. on openshift

**Comment by Dasio** [comment]
Probably fixed by https://github.com/hashicorp/vault/pull/24056

**Comment by miagilepner** [comment]
> Probably fixed by #24056
This is correct. This change will be in the next minor release, so I'm going to close this issue.
